tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/ConfidentialClientIntegrationTests.cs

```diff
@@ -57,6 +68,14 @@ public void TestInitialize()
             {
                 _keyVault = new KeyVaultSecretsProvider();
                 _ccaSecret = _keyVault.GetSecret(TestConstants.MsalCCAKeyVaultUri).Value;
+                _arlingtonCCASecret = _keyVault.GetSecret(TestConstants.MsalArlingtonCCAKeyVaultUri).Value;
+
+                if (SecretData == null)
```

Comment:
The idea of having a dictionary where the key is the Cloud and the data is some could specific detail is pretty good. But we should abstract this further because right now these tests do too much - they know how to pick between Public Cloud and Arlington data and also they know how the test logic. This breaks the Single Responsability Principle and is a code smell that there is a missing object.

Reply:
this is a temporary piece of code. the lab needs to make updates to properly return the Arlington values in the api. I have reached out to them already and this should be resolved soon. Once resolved, we can simply use the data in the lab response rather than needing to use if statements/data rows to change the test logic depending on the environment. 

Issue tracking here
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1706